dafny/lcf/def.dfy

```diff
@@ -911,151 +650,22 @@ method run(rs : RuleSet, trace : Trace) {
   }
   print "\n";
 
-  // Build tree.
-  var res := build_trace_tree(trace, 0, 0x1000_0000_0000);
-  if res.Err? {
-    print "error\n";
-    return;
-  }
-  var outcome := res.val.0;
-  if outcome.Failure? {
-    print "failure\n";
+  // Deduce theorem.
+  if |trace| == 0 {
+    print "There is no trace because it did not succeed.\n";
     return;
   }
-  if |outcome.nodes| == 0 {
-    print "no nodes";
+  if !(forall j :: 0 <= j < |trace| ==> trace[j].prop.concrete()) {
+    print "The trace is not entirely concrete.\n";
     return;
   }
-
-  print "tree:\n";
-  i := 0;
-  while i < |outcome.nodes| {
-    outcome.nodes[i].dump();
-    i := i+1;
-  }
-  print "\n";
-
-  // Deduce theorem.
-  var root := outcome.nodes[0];
-  var maybe_match := reconstruct(root, root.prop, rs);
+  print "tree:\n"; // if there was a flag that disabled printing in build_proof_tree, this line should be disabled as well
```

Comment:
What does this comment refer to?

Reply:
"// Deduce theorem" refers to the section where the theorem object is created.
"// if there was a flag, ..." is a note that in Verus port, this line will be enabled by a flag. (If the user doesn't want the tree to be printed out, they wouldn't want "tree:" to be printed.)